### ISSUE TITLE
Script to find every instance of one style and swap for another color

### ISSUE THREAD

**Issue by age3rcm** [opentoonz/opentoonz#4758]
### Is your feature request related to a problem? Please describe

i always get frustrated when i try to import a character rig to another project and cannot change its color without having it change the source project too

### Describe the solution you'd like

if i cannot bring to attention of the developers to change the palette issue, have one different approach and solution for this detail that would be crucial to get more people interested in opentoonz cut-out capabilities

with a script similar to adobe flash, where you change every instance of a color for a new one
the "find and replace" tool
![2](https://user-images.githubusercontent.com/36406840/222338922-d056ee98-45aa-4ba3-b4aa-94fab5f7dea3.png)

**Comment by RodneyBaker** [comment]
This in my estimation is ripe material for inclusion in @WolfinaBowl's Palette Editor.
It will be good to look at how we as a a community might embrace such third party efforts more fully in the future.

@Bracket-H I for one am very interested in your hacked together effort... regardless of how crude it may be.  We'll have to look at ways to encourage more experiementation in code outside of the main codebase such with the Palette Editor   mentioned above.  When we work together we can solve a lot of problems.  I'd love to see a fork of Opentoonz designed specifically for learning the ins and outs of the code; one that allows for breaking things in our quest to improve and optimize Opentoonz.

**Comment by RodneyBaker** [comment]
Yes,  it's a lot easier to do a search and replace on some color information than to juggle the UI elements.
Most definitely.

That actually would be my workaround for this report.
- Back up palette file << stomp stomp
- Open palette file in Notepad++
- Search and Replace for the color needing to be changed
- Save palette

**Comment by RodneyBaker** [comment]
Very interesting @Bracket-H
That's quite impressive.
 
Keep up the good work! :)

**Comment by RodneyBaker** [comment]
I was playing with HTML and Javascript today and put together a text replacer that allows loading of palette files.
It's very basic and is actually good for replacing text from any text file.
There is no error checking etc. and proper safeguarding to make sure we don't save over files.
Useful for a little research certainly and if I don't want to use more powerful text replacement tools it does the job.

![image](https://user-images.githubusercontent.com/2576432/226479650-a951f58e-45c7-4e6d-9aca-bf112f90643f.png)

In this image I just targetted the basic.tpl palette from stuff and note here that I enclude the name of the color in the replacement criteria.  Most styles wouldn't have that color name.  

The javascript does move through the file and replace all instances of the target text so it would replace all instances of black if that were targeted.

I added some extra fields at the bottom (currently just to contain temp text but that area could be used for other things to include displaying of the palette colors themselves.  This of course if the utility was made to focus more on colors only and not on general text replacement.

A text editor such as Notepad++ is always going to be more capable for things like this because it can use Regular Expressions (RegEx) to target all manner of things that can't be down here.
